hash_practice/exercises.py

```diff
@@ -1,19 +1,57 @@
+# Input: ["eat", "tea", "tan", "ate", "nat", "bat"],
+# Output:
+# [
+#   ["ate","eat","tea"],
+#   ["nat","tan"],
+#   ["bat"]
+# ]
+
+def sortString(string):
+    sorted_letters = sorted(string)
+    sorted_string = "".join(sorted_letters)
+    return sorted_string
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
-        Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Each sub-array will have strings which are anagrams of each other
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    temp = {}
+    array_words = []
+
+    for word in strings:
+        sorted_word = sortString(word)
+        if sorted_word in temp:
+          temp[sorted_word].append(word)
+        else:
+            temp[sorted_word] = [word]
+
+    for key in temp:
+        array_words.append(temp[key])
+    return array_words
+
+
+    
+# Input: nums = [1,1,1,2,2,3], k = 2
+# Output: [1,2]
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
-        In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        In the case of a tie it will select the first occurring element.
+        Time Complexity: O(n)

```

Comment:
Don't forget to account for the call to `sorted`, which takes O(n * log(n)) time, and thus dominates the O(n) part of this algorithm giving a final time complexity of O(n * log(n)).